Add Some Insights Endpoints (#402)
discord-userdoccers/discord-userdoccers: pages/resources/insights.mdx

# Insights

Insights are a data monitoring and analytics tool that Discord provides to help community managers understand their server's performance and how their members are interacting with it.

### Insights Object

Interval values are used to describe the time period over which the insights data is shown.

| Value | Description |
|-------|-------------|
| 1     | Daily       |
| 2     | Weekly      |
| 3     | Monthly     |

## Endpoints

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/overview">
    Get Overview
</RouteHeader>

###### Query String Params
| Field          | Type         | Description                                      |
| -------------- | ------------ | ------------------------------------------------ |
| start          | ISO8601 date | The start date for the insights data             |
| end            | ISO8601 date | The end date for the insights data               |

> | [interval](/resources/insights#interval-values) | integer | The interval for the insights data |

Reviewer: Links to enum types go in the description
Same page can just link to the anchor without full page path
The heading for this specific link doesn't exist

Author: Fixed!

Reviewer: The links weren't moved to the description

Author: OH I see what you mean 😭 sorry

Author: Alright its fixed for real I hope

  Get Guild Growth Activation by Leave Date
</RouteHeader>

###### Query String Params

| Field    | Type         | Description                                                 |
| -------- | ------------ | ----------------------------------------------------------- |
| start    | ISO8601 date | The start date for the insights data                        |
| end      | ISO8601 date | The end date for the insights data                          |
| interval | integer      | [The interval for the insights data](#aggregation-interval) |

###### Response Body

| Field         | Type         | Description                                                 |
| ------------- | ------------ | ----------------------------------------------------------- |
| day_pt        | ISO8601 date | The date of when the data was captured                      |
| days_in_guild | integer      | The number of days the user was in the guild before leaving |
| leavers       | integer      | The number of users who left the guild on this date         |

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/activation">
  Get Guild Growth Activation by Activation

> </RouteHeader>

Reviewer 2: what does this mean

> # Insights

Reviewer 2: the api routes are all under /analytics so I'm wonder if we rename this file to analytics.mdx instead? not dead set on this tho, pls bikeshed

Insights are a data monitoring and analytics tool that Discord provides to help community managers understand their server's performance and how their members are interacting with it.

> ### Insights Object

Reviewer 2: ```suggestion
```
why is this header here

> Interval values are used to describe the time period over which the insights data is shown.

Reviewer 2: ```suggestion
```
enums don't usually need descriptions unless they're very unclear

> #### Interval Values

Reviewer 2: ```suggestion
###### Aggregation Interval
```
h6 for structures/enums/flags 99% of the time, also keep it singular
(the real name for this was found in the dev portal btw if you're curious)

| Value | Name    | Description |
| ----- | ------- | ----------- |
| 1     | DAILY   | Daily       |
| 2     | WEEKLY  | Weekly      |
| 3     | MONTHLY | Monthly     |

## Endpoints

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/overview">

>   Get Guild Insight Overview

Reviewer 2: ```suggestion
  Get Guild Growth Activation Overview
```
use path for logical endpoint names

</RouteHeader>

###### Query String Params

| Field    | Type         | Description                                            |
| -------- | ------------ | ------------------------------------------------------ |
| start    | ISO8601 date | The start date for the insights data                   |
| end      | ISO8601 date | The end date for the insights data                     |
| interval | integer      | [The interval for the insights data](#interval-values) |

###### Response Body

| Field             | Type         | Description                                              |
| ----------------- | ------------ | -------------------------------------------------------- |
| day_pt            | ISO8601 date | The date of the data point                               |
| new_members       | integer      | The number of new members who joined the server          |
| new_communicators | integer      | The number of new members who communicated in the server |

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/joins">

>   Get Guild Insights by Join Date

Reviewer 2: ```suggestion
  Get Guild Growth Activation by Join Date
```
same

</RouteHeader>

###### Query String Params

| Field    | Type         | Description                                            |
| -------- | ------------ | ------------------------------------------------------ |
| start    | ISO8601 date | The start date for the insights data                   |
| end      | ISO8601 date | The end date for the insights data                     |
| interval | integer      | [The interval for the insights data](#interval-values) |

###### Response Body

| Field  | Type         | Description                                            |
| ------ | ------------ | ------------------------------------------------------ |

> | day_pt | ISO8601 date | The date of the data point                             |

Reviewer 2: what does this date actually signify, the description is quite vague

> | joins  | integer      | The number of users who joined the server on this date |

Reviewer 2: ```suggestion
| joins  | integer      | The number of users who joined the guild on this date |
```
API terminology is guild not server (same suggestion elsewhere)

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/joins-by-invite-link">

>   Get Guild Insights by invite

Reviewer 2: ```suggestion
  Get Guild Growth Activation by Invite
```
same here

</RouteHeader>

###### Query String Params

| Field    | Type         | Description                                            |
| -------- | ------------ | ------------------------------------------------------ |
| start    | ISO8601 date | The start date for the insights data                   |
| end      | ISO8601 date | The end date for the insights data                     |
| interval | integer      | [The interval for the insights data](#interval-values) |

###### Response Body

| Field       | Type         | Description                                                      |
| ----------- | ------------ | ---------------------------------------------------------------- |
| day_pt      | ISO8601 date | The date the invite link was created                             |
| invite_link | string       | The invite link used to join the server                          |
| joins       | integer      | The number of users who joined the server using this invite link |

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/joins-by-referrer">

>   Get Guild Insights by Joins by Referrer

Reviewer 2: ```suggestion
  Get Guild Growth Activation by Referrer
```
also same, I'll let you rename the rest yourself

| -------- | ------------ | ------------------------------------------------------ |
| start    | ISO8601 date | The start date for the insights data                   |
| end      | ISO8601 date | The end date for the insights data                     |
| interval | integer      | [The interval for the insights data](#interval-values) |

###### Response Body

| Field             | Type         | Description                                                               |
| ----------------- | ------------ | ------------------------------------------------------------------------- |
| day_pt            | ISO8601 date | The date the source was recorded                                          |
| discovery_joins   | integer      | The number of users who joined the server through the discovery hub       |
| invites           | integer      | The number of users who joined the server through invite links            |
| vanity_joins      | integer      | The number of users who joined the server through the server's vanity URL |
| hubs_joins        | integer      | The number of users who joined the server through the server's hubs       |

> | bot_joins         | integer      | The number of bots that joined the server                                 |

Reviewer 2: this is incorrect, don't just guess
it's the number of joins through bot means (i.e. guilds.join, see the join source type enum in guild.mdx)
same applies to below field

Returns a general overview of the amount of users who joined the guild, and the number of users who chatted in the guild.

###### Query String Params

| Field    | Type         | Description                                                 |
| -------- | ------------ | ----------------------------------------------------------- |
| start    | ISO8601 date | The start date for the insights data                        |
| end      | ISO8601 date | The end date for the insights data                          |
| interval | integer      | [The interval for the insights data](#aggregation-interval) |

###### Response Body

| Field             | Type         | Description                                             |
| ----------------- | ------------ | ------------------------------------------------------- |
| day_pt            | ISO8601 date | When the data was captured                              |
| new_members       | integer      | The number of new members who joined the guild          |
| new_communicators | integer      | The number of new members who communicated in the guild |

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/joins">

>   Get Guild Growth Activation by Join Date

Reviewer 2: ```suggestion
  Get Guild Growth Activation Joins
```

</RouteHeader>

Returns the number of users who joined the guild on a given date.

###### Query String Params

| Field    | Type         | Description                                                 |
| -------- | ------------ | ----------------------------------------------------------- |
| start    | ISO8601 date | The start date for the insights data                        |
| end      | ISO8601 date | The end date for the insights data                          |
| interval | integer      | [The interval for the insights data](#aggregation-interval) |

###### Response Body

| Field  | Type         | Description                                           |
| ------ | ------------ | ----------------------------------------------------- |
| day_pt | ISO8601 date | When the data was captured                            |
| joins  | integer      | The number of users who joined the guild on this date |

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/joins-by-invite-link">

>   Get Guild Growth Activation by Invite

Reviewer 2: ```suggestion
  Get Guild Growth Activation Joins by Invite
```

</RouteHeader>

Returns the number of users who joined the guild using an invite link.

###### Query String Params

| Field    | Type         | Description                                                 |
| -------- | ------------ | ----------------------------------------------------------- |
| start    | ISO8601 date | The start date for the insights data                        |
| end      | ISO8601 date | The end date for the insights data                          |
| interval | integer      | [The interval for the insights data](#aggregation-interval) |

###### Response Body

| Field       | Type         | Description                                                     |
| ----------- | ------------ | --------------------------------------------------------------- |
| day_pt      | ISO8601 date | The date the invite link was created                            |
| invite_link | string       | The invite link used to join the guild                          |
| joins       | integer      | The number of users who joined the guild using this invite link |

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/joins-by-referrer">

>   Get Guild Growth Activation by Referrer

Reviewer 2: ```suggestion
  Get Guild Growth Activation Joins by Referrer
```

| -------- | ------------ | ----------------------------------------------------------- |
| start    | ISO8601 date | The start date for the insights data                        |
| end      | ISO8601 date | The end date for the insights data                          |
| interval | integer      | [The interval for the insights data](#aggregation-interval) |

###### Response Body

| Field                | Type         | Description                                                       |
| -------------------- | ------------ | ----------------------------------------------------------------- |
| day_pt               | ISO8601 date | The date the referral was made                                    |
| referring_domain ^1^ | string       | The referring domain that led to the referral link being clicked  |
| joins                | integer      | The number of users who joined the guild using this referral link |

^1^ The `referring_domain` field represents what the `Referer` header contained when they joined the guild.

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/joins-by-source">

>   Get Guild Growth Activation by Sources

Reviewer 2: ```suggestion
  Get Guild Growth Activation Joins by Sources
```

| -------- | ------------ | ----------------------------------------------------------- |
| start    | ISO8601 date | The start date for the insights data                        |
| end      | ISO8601 date | The end date for the insights data                          |
| interval | integer      | [The interval for the insights data](#aggregation-interval) |

###### Response Body

| Field             | Type         | Description                                                             |
| ----------------- | ------------ | ----------------------------------------------------------------------- |
| day_pt            | ISO8601 date | The date the source was recorded                                        |
| discovery_joins   | integer      | The number of users who joined the guild through the discovery hub      |
| invites           | integer      | The number of users who joined the guild through invite links           |
| vanity_joins      | integer      | The number of users who joined the guild through the guild's vanity URL |
| hubs_joins        | integer      | The number of users who joined the guild through the guild's hubs       |

> | bot_joins         | integer      | The number of joins through bot means like guilds.join                  |

Reviewer 2: pls copy these descriptions from the join source type enum

| integration_joins | integer      | The total number of users who were added to the guild by an integration (e.g. Twitch)                                                  |
| other_joins       | integer      | The total number of users who joined the guild through an unknown source                                                               |
| total_joins       | integer      | The total number of users who joined the guild                                                                                         |

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/leavers">

>   Get Guild Growth Activation by Leave Date

Reviewer 2: ```suggestion
  Get Guild Growth Activation Leavers
```

###### Query String Params

| Field    | Type         | Description                                                 |
| -------- | ------------ | ----------------------------------------------------------- |
| start    | ISO8601 date | The start date for the insights data                        |
| end      | ISO8601 date | The end date for the insights data                          |
| interval | integer      | [The interval for the insights data](#aggregation-interval) |

###### Response Body

| Field         | Type         | Description                                                 |
| ------------- | ------------ | ----------------------------------------------------------- |
| day_pt        | ISO8601 date | When the data was captured                                  |
| days_in_guild | integer      | The number of days the user was in the guild before leaving |
| leavers       | integer      | The number of users who left the guild on this date         |

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/activation">
  Get Guild Growth Activation by Activation

> </RouteHeader>

Reviewer 2: idk why u collapsed my comment but this endpoint name doesn't make sense

>   Get Guild Growth Activation by Activation

Reviewer 2: ```suggestion
  Get Guild Growth Activation Percentages
```

| end      | ISO8601 date | The end date for the insights data                          |
| interval | integer      | [The interval for the insights data](#aggregation-interval) |

###### Response Body

| Field               | Type         | Description                                                                                     |
| ------------------- | ------------ | ----------------------------------------------------------------------------------------------- |
| day_pt              | ISO8601 date | When the data was captured                                                                      |
| new_members         | integer      | The number of new members who joined the guild on this date                                     |
| pct_communicated    | float        | The percentage of new members who sent 3 or more message in the guild or talked in a voice chat |
| pct_opened_channels | float        | The percentage of new members who opened 3 or more channels in the guild                        |

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/retention">

>   Get Guild Growth Activation by Overall Retention Rate

Reviewer 2: ```suggestion
  Get Guild Growth Activation Retention Rate
```

>   Get Guild Growth Activation by Retention Rate

Reviewer 2: ```suggestion
  Get Guild Growth Activation Retention
```

# Insights

Insights are a data monitoring and analytics tool that Discord provides to help community managers understand their guild's performance and how their members are interacting with it.

###### Aggregation Interval

| Value | Name    | Description |
| ----- | ------- | ----------- |
| 1     | DAILY   | Daily       |
| 2     | WEEKLY  | Weekly      |
| 3     | MONTHLY | Monthly     |

## Endpoints

<RouteHeader method="GET" url="/guilds/{guild.id}/analytics/growth-activation/membership">
  Get Guild Growth Activation Membership
</RouteHeader>

> Returns a list of the total number of members in the guild on the date provided.

Reviewer 2: ```suggestion
Returns a list of [growth activation membership](#growth-activation-membership-structure) objects representing the total number of members in the guild for each interval in the date range provided.
```

###### Query String Params

| Field    | Type         | Description                                                 |
| -------- | ------------ | ----------------------------------------------------------- |
| start    | ISO8601 date | The start date for the insights data                        |
| end      | ISO8601 date | The end date for the insights data                          |
| interval | integer      | [The interval for the insights data](#aggregation-interval) |
| use_t2   | boolean      | Unknown                                                     |

> ###### Response Body

Reviewer 2: ```suggestion
###### Growth Activation Membership Structure
```
